services/QuillLMS/app/services/classlink_integration/district_importer.rb

```diff
@@ -40,6 +40,14 @@ def enrollments
       @enrollments ||= EnrollmentsFetcher.run(district_oneroster_id:, district_bearer_token:, district_id:)
     end
 
+    def teacher_enrollments
+      @teacher_enrollments ||= enrollments.select { |enrollment| enrollment[:role] == 'teacher' }
+    end
+
+    def student_enrollments
+      @student_enrollments ||= enrollments.select { |enrollment| enrollment[:role] == 'student' }
+    end
+
     def run
```

Comment:
Consider moving `run` method directly below the `initialize` method, and making the rest of the methods `private`. This makes the interface of the service object clear, i.e. only `run` and `new` should ever be called. This is especially important in a class like this in which order of method calls matters (you only want them run in the order in the `run` method.